pages/design/rule-design.md

```diff
@@ -39,34 +79,64 @@ A test aspect is a part of a test subject that must be available in order to pro
 
 While most assumptions relate to the rule itself, there are some assumptions that apply at other stages of the evaluation:
 
-- It is assumed that the tested web page is the one that has to conform to WCAG 2.0 and that there is no [conforming alternative version](https://www.w3.org/TR/WCAG21/#dfn-conforming-alternate-version).
+- It is assumed that the tested web page is the one that has to conform to WCAG 2.1 and that there is no [conforming alternative version](https://www.w3.org/TR/WCAG21/#dfn-conforming-alternate-version).
 
 - It is assumed that the following technologies are accessibility supported: HTML, CSS, WAI-ARIA, ... (See also WCAG-ACT-RULES-CG's [explanation on Accessibility Support](accessibility-support.html)).
 
-## Applicability
+For more details, see [ACT Rules Format: Assumptions](https://www.w3.org/TR/act-rules-format/#assumptions).
 
-Applicability describes which (elements of) web pages should be tested using the rule. These elements are known as test targets. Applicability must be written in plain language, so that it can be used by QA testers. Applicability must rely on well defined properties of the technologies that are tested. For example, a rule may be applicable to all `video` elements, but it can not be applicable to all `object` elements used to show video, unless the term "video" is further defined.
+## Accessibility Support
 
-Finding objective definitions to use in rules can be difficult, if not outright impossible in some cases. The intent here is to ensure repeatability of the rule. Not everything in WCAG testing is entirely repeatable, but when it comes to rule applicability, this is a hard requirement.
+Sometimes a feature relevant to a rule is not fully supported by assistive technology and user agents. Include information on any known support issues for a feature relevant to the rule.
 
-For more details, see [ACT Rules Format: Applicability](https://w3c.github.io/wcag-act/act-rules-format.html#applicability)
+For more details, see [ACT Rules Format: Accessibility Support](https://www.w3.org/TR/act-rules-format/#accessibility-support).
 
-## Expectations
+## Background
+
+While optional, this provides background information relevant to the development of the rule. Any document cited in the rule should be included here, along with any applicable success criterion, understanding or technique documents, other specifications, and any other relevant reading. 
+
+For more details, see [ACT Rules Format: Background](https://www.w3.org/TR/act-rules-format/#background).
+
+## Test cases
+
+The test cases are snippets of code that help with understanding and can be used for validating implementations of the rule. There must be at least one example for **pass**, **fail** and **inapplicable** outcomes. A passing or inapplicable example should pass other applicable rules. A failing example should only fail in the one aspect being demonstrated.
```

Comment:
I disagree with the last 2 sentences "A passing or inapplicable example should pass other applicable rules. A failing example should only fail in the one aspect being demonstrated."
* Notably, we have several rules that target full documents. With these requirements, the examples must have a `alng` attribute for the `html` element, and a `title` element (or they fail the corresponding rule). I think that is confusing the reader more than helping.
* This also means that adding a new rule might require to update examples in several previous rules because there is now another applicable rule that shouldn't fail. That's just not maintenable and scalable.
* If we split rules on SC as per #1049 we will have elements that fail two rules (while currently, there is one rule per element and each element only fails one rule, which might be failing 2 SC). Requiring that failed examples only fail one rule essentially solves #1049 by saying we can't do it.

Reply:
How would you suggest we phrase it @Jym77 ? Passing and inapplicable code examples should not include accessibility fails, except possibly by omission for brevity. And failing code examples need to be clear about why the snippet is failing.

Most code snippets are actual snippets, so that is usually possible. For examples where the full page is being checked, then perhaps we should agree a default wrapping code that includes all that is needed by SC while still being brief.

Reply:
I'm not sure…
And I'm also not sure that we enforce "no accessibility fail" for Passed/Inapplicable examples.

We do enforce "do not fail the SC being tested", but I don't think we care about failing other SCs…

@WilcoFiers @kasperisager what is your opinion on Passed/Inapplicable examples failing unrelated SC?
(e.g. should https://act-rules.github.io/rules/97a4e1#inapplicable-example-2 have a `title` element and a `lang` attribute (or other tricks) in order to not fail the corresponding SC?)

Reply:
Perhaps something about code included in the snippet being an example of good practice, with allowance for brevity in relation to requirements of other SC.

@Jym77 @WilcoFiers @kasperisager I think this is the last outstanding thing to resolve.

Reply:
Yes, I'd prefer it that way.